### ISSUE TITLE
Crash with RuntimeException: StrictMode ThreadPolicy violation

### ISSUE THREAD

**Issue by noloman** [ChuckerTeam/chucker#413]
## :writing_hand: Describe the bug

I'm using Chucker in an app with Dagger 2 and Android's StrictMode with the option `StrictMode.ThreadPolicy.Builder().detectDiskWrites()`, and I get the following crash: `java.lang.RuntimeException: StrictMode ThreadPolicy violation` pointing to my this code in my `NetworkModule`:
```
@Provides
@Singleton
fun provideChuckerInterceptor(context: Context): ChuckerInterceptor = ChuckerInterceptor(context)
```
Nevertheless removing the `detectDiskWrites()` removes the crash, but I still want to have this option enabled for my app.

Thanks for the great library!

**Comment by vbuberen** [comment]
Thanks for reporting. Yes, this issue is present in 3.2.0, but it is already resolved in 3.3.0-snapshot. So feel free to use snapshot if you need to have strict mode enabled. Snapshot has more goodies besides this fix, so give it a try :)

**Comment by MiSikora** [comment]
I don't believe this is the same as #311 or #318. Unless this is about some other fix I don't recall?

I'd appreciate, @noloman, if you could share the stack trace. I'm not sure about the snippet you showed, as the only potential IO thing that happens there is DB creation/deletion and opening SharedPreferences. Also, this should not crash the app under the `StrictMode` policy.

Most likely what happens is that you read your response on the main thread. Chucker does not make any assumptions about the thread that your app uses and it will write to and read from temporary, internal files while you read your responses from the backend. In such a case it is a good thing that this was caught.

**Comment by vbuberen** [comment]
Right, I might have mixed something about issues in Chucker and some other project. Thanks for catching.

In such case yeah, we need stacktrace to understand what happens.

**Comment by koral--** [comment]
I guess that crash can be caused by `penaltyDeath()` in StrictMode configuration and the violation may come from here:
https://github.com/ChuckerTeam/chucker/blob/develop/library/src/main/java/com/chuckerteam/chucker/internal/support/AndroidCacheFileFactory.kt#L12 (in turn called from here: https://github.com/ChuckerTeam/chucker/blob/develop/library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt#L64)
It performs I/O: http://androidxref.com/6.0.0_r1/xref/frameworks/base/core/java/android/app/ContextImpl.java#createFilesDirLocked
source code slightly varies by Android version but directory creation if it does not exist is always present.

**Comment by MiSikora** [comment]
I'm not 100% sure if this applies to all devices but creating or deleting a file typically does not crash due to `penaltyDeath()`. As far as I know, it only logs it as a warning.

**Comment by noloman** [comment]
Here it is:
```
2020-08-06 09:05:29.506 15628-15628/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: me.myapp, PID: 15628
    java.lang.RuntimeException: StrictMode ThreadPolicy violation
        at android.os.StrictMode$AndroidBlockGuardPolicy.onThreadPolicyViolation(StrictMode.java:1768)
        at android.os.StrictMode$AndroidBlockGuardPolicy.lambda$handleViolationWithTimingAttempt$0$StrictMode$AndroidBlockGuardPolicy(StrictMode.java:1687)
        at android.os.-$$Lambda$StrictMode$AndroidBlockGuardPolicy$9nBulCQKaMajrWr41SB7f7YRT1I.run(Unknown Source:6)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
     Caused by: android.os.strictmode.DiskWriteViolation
        at android.os.StrictMode$AndroidBlockGuardPolicy.onWriteToDisk(StrictMode.java:1512)
        at libcore.io.BlockGuardOs.remove(BlockGuardOs.java:338)
        at libcore.io.ForwardingOs.remove(ForwardingOs.java:186)
        at android.app.ActivityThread$AndroidOs.remove(ActivityThread.java:7287)
        at java.io.UnixFileSystem.delete(UnixFileSystem.java:331)
        at java.io.File.delete(File.java:1037)
        at com.chuckerteam.chucker.internal.data.room.ChuckerDatabase$Companion.create(ChuckerDatabase.kt:22)
        at com.chuckerteam.chucker.internal.data.repository.RepositoryProvider.initialize(RepositoryProvider.kt:33)
        at com.chuckerteam.chucker.api.ChuckerCollector.<init>(ChuckerCollector.kt:32)
        at com.chuckerteam.chucker.api.ChuckerCollector.<init>(ChuckerCollector.kt:26)
        at com.chuckerteam.chucker.api.ChuckerInterceptor.<init>(ChuckerInterceptor.kt:62)
        at me.myapp.data.di.NetworkModule.provideChuckerInterceptor(NetworkModule.kt:32)
        at me.myapp.data.di.NetworkModule_ProvideChuckerInterceptorFactory.provideChuckerInterceptor(NetworkModule_ProvideChuckerInterceptorFactory.java:37)
        at me.myapp.data.di.NetworkModule_ProvideChuckerInterceptorFactory.get(NetworkModule_ProvideChuckerInterceptorFactory.java:27)
        at me.myapp.data.di.NetworkModule_ProvideChuckerInterceptorFactory.get(NetworkModule_ProvideChuckerInterceptorFactory.java:10)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at me.myapp.data.di.NetworkModule_ProvideOkHttpClientFactory.get(NetworkModule_ProvideOkHttpClientFactory.java:31)
        at me.myapp.data.di.NetworkModule_ProvideOkHttpClientFactory.get(NetworkModule_ProvideOkHttpClientFactory.java:11)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at me.myapp.data.di.ParserModule_ProvideParserFactory.get(ParserModule_ProvideParserFactory.java:27)
        at me.myapp.data.di.ParserModule_ProvideParserFactory.get(ParserModule_ProvideParserFactory.java:10)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at me.myapp.data.di.DaggerAppComponent.getFeedsViewModel(DaggerAppComponent.java:67)
        at me.myapp.data.di.DaggerAppComponent.injectNewsFragment(DaggerAppComponent.java:100)
        at me.myapp.data.di.DaggerAppComponent.inject(DaggerAppComponent.java:96)
        at me.myapp.presentation.main.NewsFragment.onAttach(NewsFragment.kt:22)
        at androidx.fragment.app.Fragment.performAttach(Fragment.java:2672)
        at androidx.fragment.app.FragmentStateManager.attach(FragmentStateManager.java:263)
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1170)
        at androidx.fragment.app.FragmentManager.addAddedFragments(FragmentManager.java:2224)
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1997)
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1953)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1849)
        at androidx.fragment.app.FragmentManager$4.run(FragmentManager.java:413)
2020-08-06 09:05:29.506 15628-15628/? E/AndroidRuntime: 	... 7 more
```

**Comment by ephemient** [comment]
Still seeing the same android.os.strictmode.DiskWriteViolation as  https://github.com/ChuckerTeam/chucker/issues/413#issuecomment-669748560 with Chucker 3.4.0 on API 30.

**Comment by MiSikora** [comment]
Yes this is expected, this issue was not addressed in `3.4.0`.

**Comment by MiSikora** [comment]
I think this issue should be fixed now with #750.

**Comment by vbuberen** [comment]
Yes, I believe so.
